### PR TITLE
Index relayed transactions

### PR DESCRIPTION
**Pull request by miiu96**
Index in elastic search also smart contract results that are generated by relayed transactions.

### CODE REVIEW

File: core/indexer/processTransactions.go [hunk 1]
```diff
@@ -118,8 +130,20 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
 	return append(convertMapTxsToSlice(transactions), rewardsTxs...)
 }
 
-func (tdp *txDatabaseProcessor) addScResultInfoInTx(scr *smartContractResult.SmartContractResult, tx *Transaction) *Transaction {
-	dbScResult := tdp.commonProcessor.convertScResultInDatabaseScr(scr)
+func findAllChildScrResults(hash string, scrs map[string]*smartContractResult.SmartContractResult) map[string]*smartContractResult.SmartContractResult {
```
Comment (sasurobert) [inline core/indexer/processTransactions.go:133]: add unit tests for intrashard and cross shard relayed smart contracts. verify how those are indexed.

Comment (miiu96) [reply core/indexer/processTransactions.go:133]: done. added a unit tests and did a lot of manual test to check if transaction are indexed correctly

File: core/indexer/common.go [hunk 2]
```diff
@@ -187,59 +187,27 @@ func (cm *commonProcessor) buildRewardTransaction(
 	}
 }
 
-func (cm *commonProcessor) convertScResultInDatabaseScr(sc *smartContractResult.SmartContractResult) ScResult {
-	decodedData := decodeScResultData(sc.Data)
+func (cm *commonProcessor) convertScResultInDatabaseScr(scHash string, sc *smartContractResult.SmartContractResult) ScResult {
```
Comment (bogdan-rosianu) [inline core/indexer/common.go:190]: into*

File: core/indexer/processTransactions.go [hunk 3]
```diff
@@ -98,6 +106,10 @@ func (tdp *txDatabaseProcessor) prepareTransactionsForDatabase(
 				}
 			}
 
+			if strings.Contains(string(transactions[hash].Data), "relayedTx") {
```
Comment (bogdan-rosianu) [inline core/indexer/processTransactions.go:109]: use core->constants.go->RelayedTransaction

File: core/indexer/processTransactions.go (hunk 1 above)
Comment (bogdan-rosianu) [reply core/indexer/processTransactions.go:133]: maybe rename it as this function also removes elements from the map received as parameter

File: core/indexer/processTransactions_test.go [hunk 4]
```diff
@@ -170,3 +170,65 @@ func TestPrepareTxLog(t *testing.T) {
...
 }
+
+func TestRelayedTransactions(t *testing.T) {
+	t.Parallel()
+
+	txHash1 := []byte("txHash1")
+	tx1 := &transaction.Transaction{
+		GasLimit: 100,
+		GasPrice: 100,
+		Data:     []byte("relayedTx@blablabllablalba"),
```
Comment (bogdan-rosianu) [inline core/indexer/processTransactions_test.go:181]: can use a prettier mock data

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**